src/core/bonded_interaction.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef BONDED_INTERACTION_H
+#define BONDED_INTERACTION_H
+enum BondedInteraction {
```

Comment:
If possible this should be an enum class. This has the advantage that it is scoped and does not pollute the namespace, and it does not decay into an int at first chance. You could then drop the BONDED_IA_ part from the names (because they are now qualified by the enum name, it is clear from the context that these are bond types. e.g. BondedInteraction::FENE ...)

I would also opt for renaming this to BondType, because this is a clearer description of what it is.